### PR TITLE
Handle other find endpoints and do not serve root URL

### PR DESCRIPTION
**Pull request by masih**
Handle `GET /cid/{cid}` and `POST /multihash` the other two flavours of
storehtinedex find APIs.

### CODE REVIEW

File: find.go [hunk 1]
```diff
@@ -15,29 +15,57 @@ import (
 func (s *server) find(w http.ResponseWriter, r *http.Request) {
 	combined := make(chan *model.FindResponse, len(s.servers))
 	wg := sync.WaitGroup{}
+
+	// Copy the original request body in case it is a POST batch find request.
+	var rb []byte
+	var err error
+	rb, err = io.ReadAll(r.Body)
```
Comment (willscott) [inline find.go:22]: do we need to limit our buffer somehow here? hopefully cloudfront would put a request body size limit in front of this already

File: find.go [hunk 2]
```diff
@@ -15,29 +15,57 @@ import (
...
+			// to the server.
+			endpoint := *r.URL
+			endpoint.Host = server.Host
+			endpoint.Scheme = server.Scheme
+			log := log.With("backend", endpoint)
+
+			// If body in original request existed, make a reader for it.
+			var body io.Reader
+			if len(rb) > 0 {
+				body = bytes.NewBuffer(rb)
```
Comment (willscott) [inline find.go:44]: we copy a new request body for each server?

File: server.go [hunk 3]
```diff
@@ -54,7 +54,8 @@ func (s *server) Serve() chan error {
 	ec := make(chan error)
 
 	mux := http.NewServeMux()
-	mux.HandleFunc("/multihash/", s.find)
+	mux.HandleFunc("/cid/", s.find)
+	mux.HandleFunc("/multihash", s.find)
```
Comment (willscott) [inline server.go:58]: will this also handle `/multihash/*`? this is not a gorilla mux, just the golang one, and i think we need both

File: find.go (hunk 2 above)
Comment (masih) [reply find.go:44]: request takes an `io.reader`. Is there a more efficient way to pass around io.reader copies across multiple goroutines?

File: find.go (hunk 1 above)
Comment (masih) [reply find.go:22]: Limited to `8 KiB`.

File: find.go (hunk 2 above)
Comment (masih) [reply find.go:44]: I am assuming the request body is re-read by each goroutine client sending the request. No?

Comment (willscott) [reply find.go:44]: you should be able to make separate readers over the same underlying byte slice https://pkg.go.dev/bytes#NewReader

**Review by willscott — APPROVED** [review]

**Pull request merged**